src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProvider.java

```diff
@@ -22,15 +22,22 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
 import com.snowflake.kafka.connector.internal.KCLogger;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
 import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.utils.Pair;
 
 /**
- * Factory that provides the streaming client(s). There should only be one provider, but it may
- * provide multiple clients if optimizations are disabled - see
- * ENABLE_STREAMING_CLIENT_OPTIMIZATION_CONFIG in the {@link SnowflakeSinkConnectorConfig }
+ * Static factory that provides streaming client(s). There should only be one provider per KC worker
+ * node, meaning that there may be multiple providers serving one connector and/or multiple
+ * connectors on one provider. If the optimization is disabled, the provider will not reuse old
+ * clients,/ see ENABLE_STREAMING_CLIENT_OPTIMIZATION_CONFIG in the {@link
+ * SnowflakeSinkConnectorConfig }
```

Comment:
I feel we should mention about map we created here. 
Its true that StreamingClientProvider is going to be per worker but props to client map is what we care more about IMO. worth mentioning that but a minor nit.

Reply:
i think the class comment should be more abstracted away from the client map for future proofing. i added a brief comment on the map object itself, what do you think about that as an alternative?